packages/notification/cardstack/service.js

```diff
@@ -0,0 +1,104 @@
+const log = require('@cardstack/plugin-utils/logger')('cardstack/notification');
+const Session = require('@cardstack/plugin-utils/session');
+const { declareInjections } = require('@cardstack/di');
+const messengerName = 'user-notification';
+
+module.exports = declareInjections({
+  encryptor: 'hub:encryptor',
+  searcher: 'hub:searchers',
+  messengers: 'hub:messengers',
+  authenticator: 'authentication:middleware'
+},
+
+class NotificationService {
+  constructor() {
+    // TODO: move these two settings into config
+    this.controllingBranch = 'master';
+  }
+
+  // TODO move this to more central place? (this is copied from authenticator middleware)
+  _tokenToSession(token) {
+    console.log("Authenticator", this.authenticator);
+    try {
+      let [sessionPayload, validUntil] = this.encryptor.verifyAndDecrypt(token);
+      if (validUntil <= Date.now()/1000) {
+        log.debug("Ignoring expired token");
+      } else {
+        return new Session(sessionPayload, this.userSearcher);
+      }
+    } catch (err) {
+      if (/unable to authenticate data|invalid key length|Not a valid signed message/.test(err.message)) {
+        log.warn("Ignoring invalid token");
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  get userSearcher() {
+    return {
+      get: (type, userId) => {
+        return this.searcher.get(this.controllingBranch, type, userId);
+      },
+      search: (params) => {
+        return this.searcher.search(this.controllingBranch, params);
+      }
+    };
+  }
+
+  async start() {
+    if (this.hasStarted) { return; }
+
+    this.hasStarted = true;
+
+    await this.messengers.getMessenger(messengerName).then(({ socketIoUrl }) => {
+      if (!socketIoUrl) {
+        log.error(`Cannot start socket.io server, no 'socketIoUrl' defined for the '${messengerName}' resouce`);
+        return;
+      }
+
+      //TODO how do we know this server is actually the host referenced in the scocket.io URL?
```

Comment:
@ef4 I'm not too sure how to solve this. In the ephemal config you can define the URL for the socket.io server. The client will use the URL to communicate to the server. But we also need to start the socket.io server correctly from this config. I'm not sure how we'll be able to tell that some particular URL (e.g. https://myapp.com:3100/socket) is actually *this* server, or whether the URL is some other server and we should skip the socket.io startup.